idaes/core/util/model_statistics.py

```diff
@@ -1034,6 +1060,19 @@ def unfixed_variables_in_activated_equalities_set(block):
     for v in variables_in_activated_equalities_set(block):
         if not v.fixed:
             var_set.add(v)
+
+    # Checks for greyboxes, and if they exist will add
+    # input and output vars to var_set if they are free
+    # inputs and outputs are defined names for greybox class and should always exist
+    for grey_box in _iter_indexed_block_data_objects(
```

Comment:
I think it would be good to have a `number_of_greybox_variables` function to do this so that users can check that separately and so that we can test the code in isolation.

Reply:
added one for variables, and unfixed variables.